### PR TITLE
fix: use existing transaction to batch insert

### LINKED ISSUE #21352: [Error] con.batchInsert is not a function (v5, Cloud)
### Required System information

Not sure how to fill this out when running in Strapi Cloud. My project is on v5.0.0 (the first non-RC version that released today, but this has been happening since at least RC 12 or so)

- Node.js version:
- NPM version:
- Strapi version:
- Database:
- Operating system:
- Is your project Javascript or Typescript:

### Describe the bug

This is only happening for me in Strapi Cloud. I contacted the Strapi Support team and they told me to report it here since they refuse to look at v5 issues.

I am not entirely sure what causes this error, but I have a Model that represents a navigation menu in a website. When trying to edit the Content for one item in that Model collection I cannot save changes because of the following error (pulled from Runtime Logs in my Strapi Cloud instance):

```
[2024-09-18 17:25:17] [2024-09-18 17:25:17.038] error: con.batchInsert is not a function
[2024-09-18 17:25:17] TypeError: con.batchInsert is not a function
[2024-09-18 17:25:17]     at /home/app/node_modules/@strapi/core/dist/services/document-service/utils/unidirectional-relations.js:52:17
[2024-09-18 17:25:17]     at /home/app/node_modules/@strapi/database/dist/index.js:6844:27
[2024-09-18 17:25:17]     at AsyncLocalStorage.run (node:async_hooks:346:14)
[2024-09-18 17:25:17]     at Object.run (/home/app/node_modules/@strapi/database/dist/index.js:2981:20)
[2024-09-18 17:25:17]     at Database.transaction (/home/app/node_modules/@strapi/database/dist/index.js:6835:27)
[2024-09-18 17:25:17]     at Module.sync (/home/app/node_modules/@strapi/core/dist/services/document-service/utils/unidirectional-relations.js:44:19)
[2024-09-18 17:25:17]     at publish (/home/app/node_modules/@strapi/core/dist/services/document-service/repository.js:237:35)
[2024-09-18 17:25:17]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[2024-09-18 17:25:17]     at async /home/app/node_modules/@strapi/database/dist/index.js:6844:21
[2024-09-18 17:25:17]     at async Array.<anonymous> (/home/app/node_modules/@strapi/i18n/dist/server/index.js:52:20)
[2024-09-18 17:25:17] [2024-09-18 17:25:17.040] http: POST /content-manager/collection-types/api::menu.menu/usek1f3lgf2b2noicfia1wmn/actions/publish (415 ms) 500

```

### Expected behavior

I should be able to save my content with no errors.

### Screenshots

Screenshot 2024-09-18 at 7 42 50 PM

### LINKED ISSUE #21395: MigrationError: Migration core::5.0.0-discard-drafts (up) failed: Original error: `con.batchInsert` is not a function
### Required System information

- Node.js version: v18.20.4
- NPM version: 10.7.0
- Strapi version: 5.0.0
- Database: PostgreSQL
- Operating system: Windows 10 21H2 (LTSC)
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When I upgrade to Strapi 5.0.0 and run Strapi for the first time, I get a migration error:

```
[2024-09-20 00:00:01.604] error: Migration core::5.0.0-discard-drafts (up) failed: Original error: con.batchInsert is not a function
MigrationError: Migration core::5.0.0-discard-drafts (up) failed: Original error: con.batchInsert is not a function
    at C:\Users\jackb\Workspace\content-manager\node_modules\umzug\lib\umzug.js:169:27
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Umzug.runCommand (C:\Users\jackb\Workspace\content-manager\node_modules\umzug\lib\umzug.js:125:20)
    at async Object.up (C:\Users\jackb\Workspace\content-manager\node_modules\@strapi\database\dist\index.js:6531:7)
    at async Object.up (C:\Users\jackb\Workspace\content-manager\node_modules\@strapi\database\dist\index.js:6555:11)
    at async Object.sync (C:\Users\jackb\Workspace\content-manager\node_modules\@strapi\database\dist\index.js:2246:9)
    at async Strapi.bootstrap (C:\Users\jackb\Workspace\content-manager\node_modules\@strapi\core\dist\Strapi.js:351:5)
    at async Strapi.load (C:\Users\jackb\Workspace\content-manager\node_modules\@strapi\core\dist\Strapi.js:315:5)
    at async Strapi.start (C:\Users\jackb\Workspace\content-manager\node_modules\@strapi\core\dist\Strapi.js:207:9)
[2024-09-20 00:00:01.606] info: Shutting down Strapi
```

### Steps to reproduce the behavior

1. Upgrade from Strapi 4.25.10.
2. Make a testing/development copy of the database from 4.25.10 and configure Strapi to use the copy.
3. Start Strapi.
4. After several minutes, the migration fails.

### Expected behavior

The migration should not fail.

### Additional context

This is probably related to https://github.com/strapi/strapi/issues/21352.

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

use existing transaction to batchInsert

potential fix for #21352 and #21395 if someone can please confirm it fixes those

### Why is it needed?

query was not using the correct knex object

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

DX-1630

fixes #21352 

fixes #21395

Fixes the (or another) problem at a different level: https://github.com/strapi/strapi/pull/21412 will investigate shortly; my feeling is we would like both that fix and this fix

### CODE REVIEW

**Comment by jackbuehner** [comment]
I can confirm that this PR also fixes the migration error in #21395.

**Comment by alexandrebodin** [comment]
@innerdvations  I see also this PR is trying to fix the batchInsert in the doc service could you take a look ? https://github.com/strapi/strapi/pull/21412

File: packages/core/core/src/services/document-service/utils/unidirectional-relations.ts [hunk 1]
```diff
@@ -98,7 +96,7 @@ const sync = async (
       });
 
       // Insert those relations into the join table
-      await con.batchInsert(joinTable.name, newRelations).transacting(trx);
+      await trx.batchInsert(joinTable.name, newRelations, 1000);
```
Comment (alexandrebodin) [inline packages/core/core/src/services/document-service/utils/unidirectional-relations.ts:99]: Is 1000 needed ? what if we have more ? should you add a check or sth ?

Comment (alexandrebodin) [reply packages/core/core/src/services/document-service/utils/unidirectional-relations.ts:99]: 👍

**Review by alexandrebodin — APPROVED** [review]

File: packages/core/core/src/services/document-service/utils/unidirectional-relations.ts (hunk 1 above)
Comment (Bassel17) [reply packages/core/core/src/services/document-service/utils/unidirectional-relations.ts:99]: https://knexjs.org/guide/utility.html#batchinsert

seems to be 1000 by default

**Review by Bassel17 — APPROVED** [review]
Looks good

**Pull request merged**